### ISSUE TITLE
Getting an error while using SimpleFindReplace.sub function 

### ISSUE THREAD

**Issue by Samson-Irudayaraj** [jorisschellekens/borb#214]
**Describe the bug**
While using SimpleFindReplace.sub function to replace a word in PDF file, the following error is thrown:

.venv\Lib\site-packages\borb\pdf\canvas\layout\layout_element.py", line 739, in paint
    assert False, f"{self.__class__.__name__} is too tall to fit inside column / page. Needed {round(cbox.get_height(), 2)} pts, only {round(cbox_available_space.get_height(), 2)} pts available."
AssertionError: Paragraph is too tall to fit inside column / page. Needed 10.80 pts, only 7.88 pts available.

**To Reproduce**
Using borb latest package installed via pip in windows machine and using Pycharm Editor
Read a PDF document.
Pass an input to the function from_word, to_word, and the PDF file to be updated
The expected output is that with SimpleFindReplace package, the PDF can be updated, and the content can be written to a new file named <original_file_name>_out.pdf
but its giving error

Steps to reproduce the behaviour:

def process_pdf_files(input_file, from_word, to_word):

    for pdf_file in input_file:

        # attempt to read a PDF
        doc: typing.Optional[Document] = None
        with open(pdf_file, "rb") as pdf_file_handle:
            doc = PDF.loads(pdf_file_handle)

        # check whether we actually read a PDF
        assert doc is not None
        # find/replace
        doc = SimpleFindReplace.sub(from_word, to_word, doc)

        #        return output_pdf

        updated_pdf_path = os.path.splitext(pdf_file)[0] + "_out.pdf"
        # Save the modified PDF to a new file (optional)
        with open(updated_pdf_path, 'wb') as output_file_pdf:
            PDF.dumps(output_file_pdf, doc)

if __name__ == '__main__':
    process_pdf_files('1843invoice.pdf', "1843", "1844")

```

**Expected behaviour**
Trying to change the word 1843 to 1844 in the 1843invoice.pdf
The expected output is that with SimpleFindReplace package, the PDF can be updated, and the content can be written to a new file named <original_file_name>_out.pdf

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: Windows 11
 - borb version [2.1.25]
- input PDF (if applicable)

**Additional context**
Add any other context about the problem here.

**Comment by jorisschellekens** [comment]
Hi @Samson-Irudayaraj,

`SimpleFindReplace` does not change the surrounding text, or the position of the surrounding text.

Let's say you had the following text in your PDF:

> In 1843 the Thames Tunnel was opened, making it the first under-water tunnel.

And you want to replace "1843" with "eighteen forty three". That would be impossible. The longer word needs to fit inside the boundaries where the shorter word fit.

Your use case is the same (although more subtle). I imagine the letter "4" is slightly wider than the letter "3".

You can change the font size of the replacement text. That might do the trick.

Kind regards,
Joris Schellekens

**Comment by Samson-Irudayaraj** [comment]
I dont think its due to the slightly wider character between 3 and 4.  I tried replacing 1843 with 1843 itself. Then also it failed.  I printed the height and width of cbox and cbox_available_space under layout_element.py line No.733 as below.  As you can clearly see, the cbox height is 10.799 where as cbox available space is 7.876.  Due to which its failing.  The cbox is getting the height and width as default values, which is leading to this problem. Can you please guide further?

cBOX 10.79999999999999960031971113   20.01600000000000041666670114
cbox avail 7.8760000000000001431355034   25.8600000000000004758693440
